debezium-connector-mariadb/src/main/java/io/debezium/connector/mariadb/MariaDbOffsetContext.java

```diff
@@ -59,7 +60,7 @@ public MariaDbOffsetContext load(Map<String, ?> offset) {
             final long binlogPosition = longOffsetValue(offset, BinlogSourceInfo.BINLOG_POSITION_OFFSET_KEY);
 
             final MariaDbOffsetContext offsetContext = new MariaDbOffsetContext(
-                    isTrue(offset, BinlogSourceInfo.SNAPSHOT_KEY),
+                    loadSnapshot((Map<String, Object>) offset),
                     isTrue(offset, SNAPSHOT_COMPLETED_KEY),
```

Comment:
IMHO it makes sense tro introduce a new `loadSnapshotMethod` as above now when the snapshot completed key is in common class.